miraclelinux/meta-debian-extended
# openssh: Remove duplicate recipe

# Purpose of pull request

The difference between meta-debian and meta-debian extended is, mainly
meta-debian-extended's recipe adds following lines.

RCONFLICTS_${PN} = "dropbear"
RCONFLICTS_${PN}-sshd = "dropbear"

Other differences are just syntax so that adding RCONFLICTS_* lines to
.bbappend file.

# Test
## How to test

1. Check openssh-sshd and dropbear cannot be installed at same time.
2. Check sshd and client work.

## Test result

Add following line to local.conf then build image.

```
IMAGE_INSTALL_append = " openssh-ssh openssh-sshd dropbear"
```

Conflict error should occur.

```
ERROR: core-image-minimal-1.0-r0 do_rootfs: Unable to install packages. Command '/home/masami/emlinux/qemuarm64-build/tmp-glibc/work/qemuarm64-emlinux-linux/core-image-minimal/1.0-r0/recipe-sysroot-native/usr/bin/apt-get  install --force-yes --allow-unauthenticated dropbear openssh-ssh openssh-sshd packagegroup-core-boot run-postinsts' returned 100:
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 openssh-sshd : Conflicts: dropbear but 2018.76-r0 is to be installed
E: Unable to correct problems, you have held broken packages.
```

Add following line to local.conf, build image, and run qemu image.

```
IMAGE_INSTALL_append = " openssh-ssh openssh-sshd"
```

Connect to ssh server.

```
masami@ubuntu1804:~/emlinux/repos/meta-debian-extended$ ssh -p 2222 root@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "cat /etc/issue ; uname -a"
Warning: Permanently added '[localhost]:2222' (ECDSA) to the list of known hosts.
EMLinux 2.4 \n \l

Linux qemuarm64 4.19.207-cip58 #1 SMP PREEMPT Thu Sep 30 04:54:59 UTC 2021 aarch64 GNU/Linux
```

ssh to localhost

```
masami@ubuntu1804:~/emlinux/repos/meta-debian-extended$ ssh -p 2222 root@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null 
Warning: Permanently added '[localhost]:2222' (ECDSA) to the list of known hosts.
Last login: Tue Oct  5 07:37:07 2021 from 10.0.2.2
root@qemuarm64:~# who
root            ttyAMA0         00:10   Oct  5 07:32:03  
root            pts/1           00:00   Oct  5 07:42:56  10.0.2.2
root@qemuarm64:~# ssh root@localhost
The authenticity of host 'localhost (127.0.0.1)' can't be established.
ECDSA key fingerprint is SHA256:9XWkzV2Cgqvj1vXxuU5D+FvqFnASetSOCg4qb6X0ZAk.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
Last login: Tue Oct  5 07:42:56 2021 from 10.0.2.2
root@qemuarm64:~# who
root            ttyAMA0         00:11   Oct  5 07:32:03  
root            pts/1           00:00   Oct  5 07:42:56  10.0.2.2
root            pts/2           00:00   Oct  5 07:43:12  127.0.0.1
```
